server/index.js

```diff
@@ -237,6 +243,31 @@ export default class Server {
     }
   }
 
+  async readBuildId () {
+    const buildIdPath = join(this.dir, '.next', 'BUILD_ID')
+    try {
+      const buildId = await fs.readFile(buildIdPath, 'utf8')
+      return buildId.trim()
+    } catch (err) {
+      if (err.code === 'ENOENT') {
+        return '-'
```

Comment:
yea, it's a bit complicated but we can notify new `buildId` to client when finishing recompilation. Maybe we can fix it later.